web/common/utils/timeSeriesUtils.ts

```diff
@@ -44,45 +35,39 @@ const mapStatForDisplay = (stat: string): string => {
   }
 };
 
-const nameHistogramStat = (matrix: TimeSeries[], stat: string, labelPrettifier?: KVMapper): TimeSeries[] => {
-  const statDisplay = mapStatForDisplay(stat);
-  matrix.forEach(ts => {
-    const labels = Object.keys(ts.labelSet)
-      .filter(k => k !== 'reporter')
-      .map(k => {
-        const val = ts.labelSet[k];
-        return labelPrettifier ? labelPrettifier(k, val) : val;
-      })
-      .join(',');
-    if (labels === '') {
-      // Ex: average // quantile 0.999 // etc.
-      ts.name = statDisplay;
-    } else {
-      // Ex: policy: average // stadium: quantile 0.999 // etc.
-      ts.name = `${labels}: ${statDisplay}`;
-    }
-  });
-  return matrix;
-};
-
-const nameTimeSeries = (matrix: TimeSeries[], chartName: string, labelPrettifier?: KVMapper): TimeSeries[] => {
-  matrix.forEach(ts => {
-    const labels = Object.keys(ts.labelSet)
-      .filter(k => k !== 'reporter')
+const nameTimeSeries = (series: TimeSeries[], labelPrettifier?: KVMapper): NamedTimeSeries[] => {
+  let hasSeveralFamilyNames = false;
+  if (series.length > 0) {
+    const firstName = series[0].labelSet[nameLabel];
+    hasSeveralFamilyNames = series.some(s => s.labelSet[nameLabel] !== firstName);
+  }
+  return series.map(ts => {
+    const name = ts.labelSet[nameLabel];
```

Comment:
If global constants are in capital letters it could be easy to check if there is a missing variable, nameLabel and statLabel perhaps can go to NAME_LABEL, STAT_LABEL for better readibility.
I think.

Reply:
The thing is that, it's true that in kiali-ui we have no consensus over how to write constants, we deliberately allow the two forms in our style guide: https://github.com/kiali/kiali-ui/blob/master/STYLE_GUIDE.adoc#variables-functions-constants.

In k-charted, at the moment it's still strictly following Basarat's guide (so a stricter version of our guide in kiali-ui), which makes no distinction between exported/not exported constants/variables (all camel case). There's other global  constants in k-charted that are camel case, and none that are upper case. So I would prefer to remain consistent, and if we think that's a problem, then change at every place.